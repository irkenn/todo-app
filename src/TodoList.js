import React, {useState} from "react";
import { v4 as uuid } from 'uuid';
import ToDo from "./Todo";
import NewToDoForm from "./NewTodoForm";


function TodoList(){
 
    const INITIAL_STATE = [];
    const [toDos, setToDos] = useState(INITIAL_STATE);
    
    const addToDo = (text) => {
        setToDos(toDos => [...toDos, {id:uuid(), text}]);
    }

    const deleteToDo = (e, id) => {
        const newToDos = toDos.filter((i) => i.id !== id);
        
        setToDos(newToDos);
    }

    return(
        <div>
            <div>
                <NewToDoForm addToDo={addToDo}/>
            </div>
            <div>
                {toDos.map(({id, text}) => <ToDo 
                                                id={id}
                                                text={text}
                                                key={id}
                                                deleteToDo={deleteToDo}/>)}
            </div>
        </div>

    )

}

export default TodoList;