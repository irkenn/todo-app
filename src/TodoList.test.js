import { render, screen } from '@testing-library/react';
import TodoList from './TodoList';

it("renders without crashing", 
  function(){
    render(<TodoList/>);
  });

it("matches the snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot()
});