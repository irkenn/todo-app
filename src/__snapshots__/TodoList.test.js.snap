// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="text"
        >
          Todo 
        </label>
        <input
          name="text"
          placeholder="write todo here"
          type="text"
          value=""
        />
        <button>
          Add todo!
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
